tests/virtualization/universal/prepare_guests.pm

```diff
@@ -46,24 +46,26 @@ sub run {
     assert_script_run "virsh net-start default || true", 90;
     assert_script_run "virsh net-autostart default",     90;
 
-    # Remove existing guests, if present. This is needed for debugging runs, when we skip the installation
-    # and it will not hurt on a fresh hypervisor
-    script_run('for i in `virsh list --name --all`; do virsh destroy $i; virsh undefine $i; done');
+    # Show all guests
+    assert_script_run 'virsh list --all';
+    wait_still_screen 1;
 
-    # Install guests
-    create_guest($_, 'virt-install') foreach (values %virt_autotest::common::guests);
+    # Disable bash monitoring, so the output of completed background jobs doesn't confuse openQA
+    script_run("set +m");
 
-    script_run 'history -a';
-    script_run('cat ~/virt-install* | grep ERROR', 30);
-    script_run('xl dmesg |grep -i "fail\|error" |grep -vi Loglevel') if (is_xen_host());
+    # Install every defined guest
+    create_guest $_, 'virt-install' foreach (values %virt_autotest::common::guests);
 
-    collect_virt_system_logs();
-}
+    ## Ensure every guest has <on_reboot>restart</on_reboot>
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        if (script_run("! virsh dumpxml $guest | grep 'on_reboot' | grep -v 'restart'") != 0) {
+            record_info("$guest bsc#1153028", "Setting on_reboot=restart failed for $guest");
+        }
+    }
 
-sub post_fail_hook {
-    my ($self) = @_;
-    collect_virt_system_logs();
-    $self->SUPER::post_fail_hook;
+    script_run 'history -a';
+    script_run('cat ~/virt-install* | grep ERROR', 30);
```

Comment:
This will be anyways checked by human, so no need for `grep`. Plus there may be important `warning` or something else…
```suggestion
    assert_script_run('cat ~/virt-install*', 30);
```

Reply:
:+1:

Reply:
Those changes were mistaken. I reverted them and also re-added the post_fail_hooks.